modelservice/src/main/java/com/expedia/adaptivealerting/modelservice/service/impl/ModelServiceImpl.java

```diff
@@ -42,26 +44,56 @@
     @Autowired
     private ModelRepository modelRepository;
 
+    @Autowired
+    private MetricRepository metricRepository;
+
     @Override
     public List<ModelDto> getModels(String metricKey) {
         return modelRepositoryCustom.findModels(metricKey);
     }
 
     @Override
-    public void addModelParams(ModelParams modelParams) {
+    public String addModelParams(ModelParams modelParams) {
 
-        Model model = new Model(modelParams.getModelUUID(), modelParams.getHyperparams());
-        Metric metric = new Metric(modelParams.getMetricKey());
-        model.getMetrics().add(metric);
-        metric.getModels().add(model);
-        modelRepository.save(model);
+        AssertUtil.notNull(modelParams.getHyperparams(), "Hyper params can't be null");
+        AssertUtil.notNull(modelParams.getMetricKey(), "Metric key can't be null");
+        AssertUtil.notNull(modelParams.getModelUUID(), "Model UUID can't be null");
+
+        Integer metricId = metricRepository.findIdByMetricKey(modelParams.getMetricKey());
+        Integer modelId = modelRepository.findIdByModelUUID(modelParams.getModelUUID());
+
+        Metric metric;
+        if (metricId == null) {
+            metric = new Metric(modelParams.getMetricKey());
+        } else {
+            metric = metricRepository.getOne(metricId);
+        }
+
+        Model model;
+        if (modelId == null) {
+            model = new Model(modelParams.getModelUUID(), modelParams.getHyperparams());
+        } else {
+            model = modelRepository.getOne(modelId);
+        }
+
+        if (metricId == null || modelId == null) {
+            model.getMetrics().add(metric);
+            metric.getModels().add(model);
+            modelRepository.save(model);
+            return "Model params saved successfully";
+        } else {
+            return "Metric-Model mapping already exists";
+        }
```

Comment:
These return values are strange for a model service. Normally we would not use display text as a return value for a successful operation as human-friendly display text is a UI concern, not a service concern. I would expect to see void return type here and an exception if anything goes wrong.